src/rust/ensogl/src/display/shape/text/text_field/cursor.rs

```diff
@@ -147,9 +152,12 @@ pub struct CursorNavigation<'a> {
 
 impl<'a> CursorNavigation<'a> {
     /// Creates a new CursorNavigation with defaults.
-    pub fn default(content:&'a mut TextFieldContent) -> Self {
+    pub fn default
+    ( content         : &'a mut TextFieldContent
+    , text_field_size : Vector2<f32>
+    , scroll_position : Vector2<f32>) -> Self {
         let selecting     = default();
-        Self {content,selecting}
+        Self {content,selecting,text_field_size,scroll_position}
```

Comment:
I wonder if it should be still called "default" function. I meant to have a function which as all fields in some default state except the content (which cannot be defaulted), but this looks silly when you have 4 fields and only one can be actually defaulted :)

Reply:
This is true. I think it makes sense to pass `TextField` to it instead of these three different `TextField`'s properties separately.